src/Trees/Helpers.hs

```diff
@@ -142,8 +145,21 @@ numOfOpsInFormula (Brackets f) = numOfOpsInFormula f
 numOfOpsInFormula (Assoc _ f1 f2) = 1 + numOfOpsInFormula f1 + numOfOpsInFormula f2
 
 numOfUniqueBinOpsInSynTree :: SynTree BinOp c -> Integer
-numOfUniqueBinOpsInSynTree x = fromIntegral (length (nubOrd (ops x)))
-  where ops :: SynTree BinOp c -> [BinOp]
-        ops (Leaf _) = []
-        ops (Not f) = ops f
-        ops (Binary op s1 s2) = concat [[op], ops s1, ops s2]
+numOfUniqueBinOpsInSynTree = fromIntegral . length . collectUniqueBinOpsInSynTree
+
+binOp :: SynTree BinOp a -> Maybe BinOp
+binOp (Binary op _ _) = Just op
+binOp _ = Nothing
+
+subTrees :: SynTree BinOp a -> [SynTree BinOp a]
+subTrees (Leaf _) = []
+subTrees (Not x) = [x]
+subTrees (Binary _ l r) = [l, r]
```

Comment:
Hier (insbesondere angesichts der Verwendung oben in `src/Tasks/ComposeFormula/Quiz.hs`) fände ich etwas wie
```haskell
bothKids :: SynTree BinOp a -> (SynTree BinOp a, SynTree BinOp a)
bothKids (Binary _ l r) = (l, r)
bothKids _ = error "impossible to land here"
```
besser.

Und generell bei solchen Baumverarbeitungsfunktionen: Wenn der Typ der Operatoren nicht relevant ist, dann kann diese nützliche Polyorphie-Information im Typ reflektiert werden. Hier also etwa durch:
```haskell
bothKids :: SynTree o a -> (SynTree o a, SynTree o a)
```

Reply:
Habe ich entsprechend geändert.